editor/src/clj/editor/cljfx_form_view.clj

```diff
@@ -423,30 +424,66 @@
   {:set-ui-state (update-in ui-state state-path dissoc :edit)
    :dispatch (assoc on-edited :fx/event {:index index :item event})})
 
-(defmethod handle-event :add-list-element [{:keys [value
+(defn- absolute-or-maybe-proj-path
+  ^String [^File file workspace in-project]
+  (if in-project
+    (workspace/as-proj-path workspace file)
+    (.getAbsolutePath file)))
+
+(defn query-added-list-elements-or-dialog-type [element ^Event event workspace project]
```

Comment:
So the return contract of this function is "either `:directory-not-in-project`, `:file-not-in-project` or a list of values", and the first 2 options are only used to show error dialogs. I understand the rationale, but I'm not very happy about it. I think it would be nicer if it just showed the dialog and returned `nil` instead, but I also see that this function is unlikely to be used anywhere else, so it's fine :)